python/cugraph/structure/renumber.py

```diff
@@ -63,3 +66,105 @@ def renumber(source_col, dest_col):
+    - The source and destination column names cannot be the same or overlap.
+    - The data type order needs to be the same between source and destination
+        columns. This is due to the two sets being merged to create a single
+        list of all possible values
+
+    Input Parameters
+    ----------
+    df : cudf.DataFrame
+        The dataframe containing the source and destination columans
+    source_cols_names : List
+        This is a list of source column names
+    dest_cols_names : List
+        This is a list of destination column names
+
+    Returns
+    ---------
+    src_ids : cudf.Series
+        The new source vertex IDs
+    dst_ids : cudf.Series
+        The new destination vertex IDs
+    numbering_df : cudf.DataFrame
+        a dataframe that maps a vertex ID to the unique
+
+
+    Examples
+    --------
+    >>> gdf = cudf.read_csv('datasets/karate.csv', delimiter=' ',
+    >>>                   dtype=['int32', 'int32', 'float32'], header=None)
+
+    >>> source_col, dest_col, numbering_map =
+    >>>    cugraph.renumber_from_cudf(gdf, ["0"], ["1"])
+    >>>
+    >>> G = cugraph.Graph()
+    >>> G.add_edge_list(source_col, dest_col, None)
+    """
+    if len(source_cols_names) == 0:
+        raise ValueError('Source column list is empty')
+
+    if len(dest_cols_names) == 0:
+        raise ValueError('Destination column list is empty')
+
+    if len(source_cols_names) != len(dest_cols_names):
+        raise ValueError(
+            'Source and Destination column lists are not the same size')
+
+    # ---------------------------------------------------
+    # get the source column names and map to indexes
+    _src_map = OrderedDict()
+    for i in range(len(source_cols_names)):
+        _src_map.update({source_cols_names[i]: str(i)})
+
+    _tmp_df_src = _df[source_cols_names].rename(_src_map).reset_index()
+
+    # --------------------------------------------------------
+    # get the destination column names and map to indexes
+    _dst_map = OrderedDict()
+    for i in range(len(dest_cols_names)):
+        _dst_map.update({dest_cols_names[i]: str(i)})
+
+    _tmp_df_dst = _df[dest_cols_names].rename(_dst_map).reset_index()
+
+    _vals = list(_src_map.values())
+
+    # ------------------------------------
+    _s = _tmp_df_src.drop('index').drop_duplicates()
+    _d = _tmp_df_dst.drop('index').drop_duplicates()
+
+    _tmp_df = cudf.concat([_s, _d])
+    _tmp_df = _tmp_df.drop_duplicates().reset_index().drop('index')
+
+    if len(_tmp_df) > np.iinfo(np.int32).max:
+        raise ValueError('dataset is larger than int32')
+
+    _tmp_df['id'] = _tmp_df.index.astype(np.int32)
+
+    del _s
+    del _d
+
+    _src_ids = _tmp_df_src.merge(
+        _tmp_df, on=_vals, how='left').drop(_vals).sort_values(by='index')
+
+    _dst_ids = _tmp_df_dst.merge(
+        _tmp_df, on=_vals, how='left').drop(_vals).sort_values(by='index')
+
+    return _src_ids['id'], _dst_ids['id'], _tmp_df
```

Comment:
the returned numbering_df has 'id' column which can maybe be dropped after the merge, as it is same as index values

Reply:
true, but then it would be tough to be a bulk `merge` to translate results back to columns.

Reply:
I'm not sure what you mean by merge to translate results back to columns. Are you referring to the merge to get _src_ids_ and _dst_ids_?
Dropping ids by returning _tmp_df.drop('ids') at 170 could be more user friendly. Or if it has some implications later, we could leave it as is.

Reply:
without the `id` column you just have the list of unique column values.  To un-renumber the data, you need to have a field that can be used join the graph results data with the numbering dataframe.  

unnumbered = results.merge(number_df, left_on='`vertex`, right_on='id', how='left')

Reply:
the iterative way we do it for testing is okay for testing, but not for performance.  and if the number datafrane order is changed, then it won't work.
for i in len(results):
    col = numbering[results['vertex'][i]

Reply:
agreed, for unrenumbering later, the merge on id option is better suited for multi-column